__tests__/commands/add.js

```diff
@@ -873,3 +874,55 @@ test.concurrent('installing with --pure-lockfile and then adding should keep bui
     expect(await fs.exists(path.join(config.cwd, 'node_modules', 'package-a', 'temp.txt'))).toBe(true);
   });
 });
+
+if (!isCI) {
```

Comment:
These tests would never run on CI which would make them effectively useless at catching potential issues. You should simulate this behavior and test the output.